rainier-compute/src/main/scala/com/stripe/rainier/compute/Cholesky.scala

```diff
@@ -0,0 +1,86 @@
+package com.stripe.rainier.compute
+
+/*
+This represents an NxN matrix A
+by the  N(N+1)/2 packed lower-triangular elements
+of its Cholesky decomposition L
+(where A = LL*)
+*/
+case class Cholesky(packed: Vector[Real]) {
+    val rank: Int = Cholesky.rank(packed.size)
+    
+    //log(det(A))
+    def logDeterminant: Real = Real.sum(diagonals.map(_.log)) * 2
```

Comment:
I'd consider making this a `lazy val`

Reply:
I think it's actually unlikely to get used more than once (remembering that this is constructing Reals and so run at DAG-building time, not in a tight loop)